Change redirect 404 to rewrite

matematik7 opened the pull request:
Change redirect 404 to rewrite.
When I access path like /dashboard/6, I should be served `index.html` not redirect to it, so that I can refresh the page and stay on it.

m-terel reviewed (commented):
A new impl is similar to the previous. Please tell, why wasn't it just changed instead of replacing it with a new file? It just would be easier to review the diff.

matematik7 commented:
@m-terel I moved the file and changed the name since it's not a redirect anymore but a rewrite of url. Usually git would handle movement of file just fine if there are enough similarities, but seems there were too many changes.
I don't think it's a big deal, since there are only a few lines that are actually the same as before. There are quiet a few changes since now it also doesn't propagate header changes until status code is sent.

m-terel commented on server/api/middleware/rewrite404_test.go:
```diff
@@ -0,0 +1,54 @@
+		Path string
+	}{
+		{
+			Path: "/testpath",
```
nit: maybe a better naming for clarity would be good, `"testpath"->"not_found_path"`?

matematik7 replied:
Good idea. Done.

m-terel reviewed (approved):
I see. They still have a lot in common. If the file was renamed, but it just the git shows it as a new file, then it's ok. Good to go.

m-terel reviewed (approved):
I see. They still have a lot in common. If the file was renamed, but it just the git shows it as a new file, then it's ok. Good to go.
